### PR TITLE
feat(RHINENG-10487): Convert MQ service to batch inserts/updates

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-10487](https://issues.redhat.com/browse/RHINENG-10487).
It changes the implementation of the MQ service so that it inserts/updates in batches. Some restructuring was necessary in order to separate the DB operations from the MQ event production. 

I also had to make a number of changes to the related unit tests, as the method signatures and functionality changed.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [x] Tests: edge cases
- [x] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/config.py [hunk 1]
```diff
@@ -276,6 +276,8 @@ def __init__(self, runtime_environment):
         self.script_chunk_size = int(os.getenv("SCRIPT_CHUNK_SIZE", "500"))
         self.rebuild_events_time_limit = int(os.getenv("REBUILD_EVENTS_TIME_LIMIT", "3600"))  # 1 hour
         self.sp_authorized_users = os.getenv("SP_AUTHORIZED_USERS", "tuser@redhat.com").split()
+        self.mq_db_batch_max_messages = int(os.getenv("MQ_DB_BATCH_MAX_MESSAGES", "50"))
```
Comment (chambridge) [inline app/config.py:279]:
```suggestion
        self.mq_db_batch_max_messages = int(os.getenv("MQ_DB_BATCH_MAX_MESSAGES", "1"))
```
Nit, but to be consistent with the clowdapp file should this be `1`. Don't have to change it now, but noticed the inconsistency where a dev might have assumed the in code default might be used.

File: app/instrumentation.py [hunk 2]
```diff
@@ -232,8 +232,8 @@ def log_add_host_attempt(logger, input_host):
     )
 
 
-def log_add_update_host_succeeded(logger, add_result, host_data, output_host):
-    metrics.add_host_success.labels(add_result.name, host_data.get("reporter", "null")).inc()  # created vs updated
+def log_add_update_host_succeeded(logger, add_result, output_host):
+    metrics.add_host_success.labels(add_result.name, output_host["reporter"]).inc()  # created vs updated
```
Comment (chambridge) [inline app/instrumentation.py:236]: Do we always get a reporter back now. Does it resolve to "null" or "None" now? Just trying to understand why the default "null" if the key wasn't there is better, or not an issue.

File: app/queue/queue.py [hunk 3]
```diff
@@ -219,19 +237,14 @@ def update_system_profile(host_data, platform_metadata, operation_args={}):
...
-            staleness = get_staleness_obj(identity)
-            output_host, host_id, insights_id, update_result = host_repository.update_system_profile(
-                input_host, identity, staleness_timestamps, staleness=staleness
-            )
-            log_update_system_profile_success(logger, output_host)
-            payload_tracker_processing_ctx.inventory_id = output_host["id"]
-            return output_host, host_id, insights_id, update_result
+            output_host, update_result = host_repository.update_system_profile(input_host, identity)
+            success_logger = partial(log_update_system_profile_success, logger)
+            return output_host, update_result, identity, success_logger
```
Comment (chambridge) [inline app/queue/queue.py:247]: Just for awareness, are we updating payload tracker when the save occurs, but not the commit? Do we need to do anything ahead of a stage deploy to confirm that payload tracker works as expected?

**Review by chambridge — COMMENTED** [review]
Had a couple minor questions, but otherwise happy to approve afterwards.

File: app/instrumentation.py (hunk 2 above)
Comment (kruai) [reply app/instrumentation.py:236]: Pretty sure we always do, but I'll switch it to use the `.get()` just to be safe.

**Review by chambridge — APPROVED** [review]
LGTM

**Pull request merged**